### PR TITLE
Bmd 5314 servicediscovery

### PR DESCRIPTION
**Pull request by Maciej-Gierwatowski**

### CODE REVIEW

**Review by bmwosz — APPROVED** [review]
looks ok

File: main.tf [hunk 1]
```diff
@@ -348,6 +348,13 @@ resource "aws_iam_policy" "kms_key_access_policy" {
   policy = data.aws_iam_policy_document.kms_key_policy_document.json
 }
 
+# create service discovery
+resource "aws_service_discovery_private_dns_namespace" "default" {
+  name        = "${module.label.id}.local"
```
Comment (bmtoml) [inline main.tf:353]: shouldn't this be dependent on the service discovery enabled flag?

File: main.tf [hunk 2]
```diff
@@ -366,33 +373,35 @@ locals {
...
-    "kms_key_alias_name"              = module.kms_key.alias_name
-    "kms_key_arn"                     = module.kms_key.key_arn
-    "kms_key_name"                    = module.kms_key.key_id
-    "kms_key_access_policy_arn"       = aws_iam_policy.kms_key_access_policy.arn
+    "kms_key_alias_arn"              = module.kms_key.alias_arn
+    "kms_key_alias_name"             = module.kms_key.alias_name
+    "kms_key_arn"                    = module.kms_key.key_arn
+    "kms_key_name"                   = module.kms_key.key_id
+    "kms_key_access_policy_arn"      = aws_iam_policy.kms_key_access_policy.arn
+    "service_discovery_namespace_id" = aws_service_discovery_private_dns_namespace.default.id
```
Comment (bmtoml) [inline main.tf:404]: what happens if SD is not enabled?

File: main.tf (hunk 1 above)
Comment (Maciej-Gierwatowski) [reply main.tf:354]: fixed

File: main.tf (hunk 2 above)
Comment (Maciej-Gierwatowski) [reply main.tf]: fixed

**Pull request merged**